cli/args/mod.rs

```diff
@@ -513,7 +625,27 @@ impl CliOptions {
     self.flags.env_file.as_ref()
   }
 
-  pub fn resolve_main_module(&self) -> Result<&ModuleSpecifier, AnyError> {
+  fn resolve_main_module_with_resolver_if_bare(
+    &self,
+    raw_specifier: &str,
+    resolver: Option<impl MainModuleResolver>,
+    default_resolve: impl Fn() -> Result<ModuleSpecifier, AnyError>,
+  ) -> Result<ModuleSpecifier, AnyError> {
+    match resolver {
+      Some(resolver) if !raw_specifier.starts_with(['/', '.', '\\']) => {
```

Comment:
What happens with an absolute windows path like `C:\something.ts`? Not sure we have a test for that.

Reply:
always forget windows 😅, will update